platform/app-manager/app-manager.js

```diff
@@ -18,71 +21,109 @@ mongoClient.connect(mongoUrl, { useNewUrlParser: true }, function(err, client) {
     db = client.db(appsDb);
 });
 
-// Create logs directory if not present
+// Create logs directory for apps if not present
 fs.ensureDirSync(`${__dirname}/logs`);
 /**
  * This function saves the app info to the database
- * @param {string} appPath application path
+ * @param {string} appName application name
  * @param {string} topic mqtt topic
  * @param {string} pid application pid
- * @returns {string}
+ * @returns {int}
  */
-function saveAppInfoToDB(appPath, topic, pid) {
-    let appId;
+function saveAppInfoToDB(appName, topic, pid) {
     try {
-        let result = db.collection(appsInfoCollection).insertOne( { "appName": appPath, "topic": topic, "pid": pid });
-        appId = result["insertedId"];
+        let result = db.collection(appsInfoCollection)
+                        .update({ "_id": new ObjectId(topic) }, {
+                            "pid": pid,
+                            "name": appName
+                        });
+        if("writeConcernError" in result)
+            throw result["writeConcernError"];
+        console.log(`Deployed successfully!`);
+        console.log(`  time:   ${new Date().toISOString()}`);
+        console.log(`  app:    ${appName}`);
+        console.log(`  topic:  ${topic}`);
+        console.log(`  pid:    ${pid}`);
     } catch (err) {
+        console.error(`deploy ${appName} failed.`);
         console.error(err);
-    }
-    return appId;
+        return 1;
+    };
+    return 0;
 }
 
 // apps stores process, topic, pid, and path
 const apps = {};
 
 /**
  * This function generates the topic for new coming app.
- * TODO: use app name or other?
+ * the topic is the id generated by mongodb
  * @param {string} appName
  * @returns {string}
  */
-function getTopic(appName) {
-    let topic = appName;
-    return topic;
+async function getTopic(appName) {
```

Comment:
We need to talk about the topic generation

Reply:
How about this?
[reference](https://nodejs.org/dist/latest-v12.x/docs/api/crypto.html#crypto_class_hash)
```node
const crypto = require('crypto');
const hash = crypto.createHash('sha256');

hash.update('/path/of/the/app.js');
console.log(hash.digest('hex'));
// Prints:
//   6a2da20943931e9834fc12cfe5bb47bbd9ae43489a30726962b576f4e3993e50
```

Reply:
use a more compact hash function